### PR TITLE
Check if can compile semgrep with 5.2.0~beta2

### PR DESCRIPTION
**Pull request by aryx**
test plan:
wait for green CI check

### CODE REVIEW

**Comment by aryx** [comment]
problem with setup-ocaml apparently. I've reported it here:
https://github.com/ocaml/setup-ocaml/issues/778

File: .github/workflows/build-test-core-x86-ocaml5.jsonnet [hunk 1]
```diff
@@ -6,8 +6,7 @@ local actions = import 'libs/actions.libsonnet';
 local semgrep = import 'libs/semgrep.libsonnet';
 
 //TODO: currently fails because of conflicts with ocamlformat and jsoo
-//TODO: local opam_switch = '5.2.0~alpha1';
-local opam_switch = '5.1.0';
+local opam_switch = 'ocaml-base-compiler.5.2.0~beta2';
```
Comment (aryx) [inline .github/workflows/build-test-core-x86-ocaml5.jsonnet:11]: Add a comment explaining why we can't use 5.2.0~beta2; just add a link to the issue

File: .github/workflows/build-test-core-x86-ocaml5.jsonnet [hunk 2]
```diff
@@ -22,6 +21,10 @@ local job = {
       uses: 'ocaml/setup-ocaml@v2',
       with: {
         'ocaml-compiler': opam_switch,
+        'opam-repositories': |||
+            default: https://github.com/ocaml/opam-repository.git
+            beta: https://github.com/ocaml/ocaml-beta-repository.git
```
Comment (aryx) [inline .github/workflows/build-test-core-x86-ocaml5.jsonnet:30]: Same here, add a comment explaining why we need those special settings.

File: dev/required.opam [hunk 3]
```diff
@@ -8,5 +8,6 @@ bug-reports: "n/a"
 synopsis: "OCaml dependencies needed in CI"
 
 depends: [
-  "ocamlformat" {= "0.26.1" }  # used by the pre-commit hook
+  # commented out since it doesn't support 5.2
```
Comment (aryx) [inline dev/required.opam]: we probably need to move it to optional.opam maybe? @mjambon maybe knows better.

**Review by aryx — COMMENTED** [review]
Nice!

File: dev/required.opam (hunk 3 above)
Comment (ajbt200128) [reply dev/required.opam]: check latest commit, cleaned it up so it only installs if ocaml < 5.0

File: .github/workflows/build-test-core-x86-ocaml5.jsonnet (hunk 1 above)
Comment (ajbt200128) [reply .github/workflows/build-test-core-x86-ocaml5.jsonnet:11]: will do

**Comment by ajbt200128** [comment]
was requested to review this but contributed, so will let someone else. @mjambon would appreciate your review since we do some testo stuff and I wouldn't be surprised if I did it in an ugly way

**Comment by ajbt200128** [comment]
tested ocamlformat required.opam stuff by:
```bash
# on ocaml 4.X.X switch
opam uninstall ocamlformat
make install-opam-deps
# ocamlformat listed as install goal
opam switch 5.1.1 5.1.1
make install-opam-deps
# succeeds
```

**Comment by aryx** [comment]
Sure, I'll take over. Thx a lot @ajbt200128 for the fixes.

File: dev/required.opam [hunk 4]
```diff
@@ -8,5 +8,6 @@ bug-reports: "n/a"
 synopsis: "OCaml dependencies needed in CI"
 
 depends: [
-  "ocamlformat" {= "0.26.1" }  # used by the pre-commit hook
+  # OCamlFormat doesn't support OCaml 5 https://github.com/ocaml-ppx/ocamlformat/issues/2524
+  (("ocamlformat" {= "0.26.1" } & "ocaml" {< "5.0"}) | "ocaml" {>= "5.0"}) # used by the pre-commit hook
```
Comment (aryx) [inline dev/required.opam:12]: nice, I didn't think about that. Perfect!

**Review by neuroo — APPROVED** [review]

**Pull request merged**